activesupport/lib/active_support/core_ext/numeric/conversions.rb

```diff
@@ -118,9 +118,16 @@ def to_formatted_s(format = :default, options = {})
     end
   end
 
-  [Float, Fixnum, Bignum, BigDecimal].each do |klass|
+  [Float, Integer, Fixnum, Bignum, BigDecimal].each do |klass|
     klass.send(:alias_method, :to_default_s, :to_s)
 
+    # Ruby 2.4+ unifies Fixnum & Bignum into Integer.
+    if Integer == Fixnum
+      next if klass == Fixnum || klass == Bignum
+    else
+      next if klass == Fixnum
```

Comment:
That makes sense. I applied to patch and rebased the branch. I'll merge on green.